src/commands/registries/azure/viewAzureProperties.ts

```diff
@@ -3,17 +3,15 @@
  *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { IActionContext, openReadOnlyJson } from "@microsoft/vscode-azext-utils";
+import { IActionContext, contextValueExperience, openReadOnlyJson } from "@microsoft/vscode-azext-utils";
 import { ext } from "../../../extensionVariables";
-import { AzureRegistryTreeItem } from "../../../tree/registries/azure/AzureRegistryTreeItem";
-import { AzureTaskRunTreeItem } from "../../../tree/registries/azure/AzureTaskRunTreeItem";
-import { AzureTaskTreeItem } from "../../../tree/registries/azure/AzureTaskTreeItem";
-import { registryExpectedContextValues } from "../../../tree/registries/registryContextValues";
+import { AzureRegistry } from "../../../tree/registries/Azure/AzureRegistryDataProvider";
+import { UnifiedRegistryItem } from "../../../tree/registries/UnifiedRegistryTreeDataProvider";
 
-export async function viewAzureProperties(context: IActionContext, node?: AzureRegistryTreeItem | AzureTaskTreeItem | AzureTaskRunTreeItem): Promise<void> {
+export async function viewAzureProperties(context: IActionContext, node?: UnifiedRegistryItem<AzureRegistry>): Promise<void> {
     if (!node) {
-        node = await ext.registriesTree.showTreeItemPicker<AzureRegistryTreeItem>(registryExpectedContextValues.azure.registry, context);
+        node = await contextValueExperience(context, ext.azureRegistryDataProvider, { 'include': 'azureContainerRegistry' });
```

Comment:
What happens when:

1. You are logged in to Azure, but the Azure reg provider is not connected
2. You are not logged in nor is the Azure reg provider connected

and you execute this from the command palette?

Reply:
1. it throws an error: Cannot read properties of undefined (reading 'label'). I think this is due to the tree not being registered
2. same thing happens ^

Reply:
this also happens in other azure commands as well

Reply:
Well, we definitely don't want it to throw an ugly error like that. We should change it so that it throws an error along the lines of "No Azure registries are connected."

Reply:
actually the tree is being registered before the command is executed, how can this be the case

Reply:
As in, the Azure RDP is being registered, or _connected_ before the command is executed? I would expect it to be registered during extension activation regardless of whether it is connected or not.

Reply:
i looked into this a little more, the `node` in  `node = await contextValueExperience(context, ext.azureRegistryDataProvider, { 'include': 'azureContainerRegistry' });` will actually become the wrappedItem and therefore `node.wrappedItem` is `undefined`. Interestingly, if you pass in `ext.registryTree`, everything will work. 

This is making me want to really work on `registryExperience()` so that we can pass in `ext.registryTree` and some extra arguments (in this case 'azure') to make the picks only show azure in this case

Reply:
as we discussed offline, your approach makes more sense, this commit should address the issue https://github.com/microsoft/vscode-docker/pull/4026/commits/f84bdaa8ccbb0af4d225b70628a17b7e27b7e6b5